src/YoutubeAtom.tsx

```diff
@@ -64,8 +72,172 @@ type YoutubeAtomType = {
     title?: string;
     duration?: number; // in seconds
     origin?: string;
+    gaEventEmitter?: (event: VideoEventKey) => void;
+    ophanEventEmitter?: (event: VideoEventKey) => void;
+};
+
+let progressTracker: NodeJS.Timeout | null;
+
+// make sure that only 1 event has been sent per video
+let has25PercentEventBeenDispatched = false;
+let has50PercentEventBeenDispatched = false;
+let has75PercentEventBeenDispatched = false;
+export const onPlayerStateChangeAnalytics = ({
+    e,
+    hasUserLaunchedPlay,
+    setHasUserLaunchedPlay,
+    gaEventEmitter,
+    ophanEventEmitter,
+    getCurrentTime,
+    getDuration,
+}: {
+    e: YoutubeStateChangeEventType;
+    hasUserLaunchedPlay: boolean;
+    setHasUserLaunchedPlay: (userLaunchEvent: boolean) => void;
+    gaEventEmitter?: (event: VideoEventKey) => void;
+    ophanEventEmitter?: (event: VideoEventKey) => void;
+    getCurrentTime: () => number;
+    getDuration: () => number;
+}) => {
+    /** YouTube API
+            -1 (unstarted)
+            0 (ended)
+            1 (playing)
+            2 (paused)
+            3 (buffering)
+            5 (video cued)
+        **/
+    switch (e.data) {
+        // playing
+        case 1: {
+            // need to make sure user is not resuming
+            if (!hasUserLaunchedPlay) {
+                gaEventEmitter && gaEventEmitter('play');
+                ophanEventEmitter && ophanEventEmitter('play');
+            }
+            setHasUserLaunchedPlay(true);
+
+            progressTracker = setInterval(() => {
+                if (!gaEventEmitter)
+                    // eslint-disable-next-line no-console
+                    console.error(`GA function is not available`);
```

Comment:
Do we want to throw this errors on PROD? maybe we want to check `process.ENV` is not prod (well, if you don't swap to have an array of eventEmitters that is!)